src/github.com/getlantern/balancer/balancer.go

```diff
@@ -49,7 +68,25 @@ func New(dialers ...*Dialer) *Balancer {
 // remaining Dialers until it either manages to connect, or runs out of dialers
 // in which case it returns an error.
 func (b *Balancer) DialQOS(network, addr string, targetQOS int) (net.Conn, error) {
-	dialers := b.dialers
+	var dialers []*dialer
+
+	// Checking destination port.
+	_, port, _ := net.SplitHostPort(addr)
+
+	// Are we attempting to connect to port 80 (plain HTTP)?
```

Comment:
I can't remember offhand why we opened 5228, but it was basically because people were using it.

I don't think we're making any assumptions about the actual data being run over any of these ports are we? In fact we don't add any extra encryption to any traffic other than always encrypting connections between the Lantern client and any proxy it hits (with that resulting in TLS over tunneled TLS in the case of connections that are already encrypted).

I guess I'm a little confused. The theoretical attack here is a mixed content attack. Most browsers actually protect against that for JavaScript (see https://qualys.jive-mobile.com/#jive-document?content=%2Fapi%2Fcore%2Fv2%2Fposts%2F2400). The attack is specifically when browsers load HTTPS pages that include HTTP content. Let's say that's a script (let's imagine the browser isn't protecting against script injections). In that case, though, the javascript would be loaded over HTTP and port 80 typically, so if we proxied port 80 we'd still be vulnerable to that.

I.e. only proxying port 80 on our servers wouldn't mitigate the attack since the attack is typically over port 80.

What am I missing?